### PR TITLE
Pr128 2

### PR DESCRIPTION
**Pull request by calebkrauter**
PR 128 had merge conflicts that needed resolving off of github. I copied @B1GD3VD0G 's PR into this current branch PR128-2 so that it could be rebased into a copy of my development branch. This has a couple of adjustments that are needed however the dialog functionality is now present and working in the village in this PR. There may also be a glitch where cobblestone is not displaying properly. 

The healthbar is still not displaying but that was missing in Nathan's changes as well.

The PR this is based on where the conflicts arose and were fixed is from: https://github.com/GoodBadChad/good-bad-chad-br/pull/128#issue-2135335211

The reason that the conflicts were not simply just resolved in that PR and instead a rebasing was done is because I ran into an issue as found in the photo below. I was able to attempt a conflict resolution at one point however after publishing the resolution in the branch the resolution/changes did not appear on the remote branch and I ran into this issue.
Screenshot 2024-02-17 at 12 08 55 AM

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -74,7 +72,7 @@
     <script type="text/javascript" src="./js/classes/entities/midground/enemies/village/snake.js"></script>
 
     <script type="text/javascript" src="./js/classes/entities/midground/enemies/snowy/yeti.js"></script>
->
+    >
```
Comment (tclaar) [inline index.html:75]: Random angle bracket?

File: .gitignore
Comment (tclaar) [inline .gitignore:1]: Thanks for finally adding this lol

File: js/classes/asset_manager.js [hunk 2]
```diff
@@ -183,67 +183,67 @@ class AssetManager {
...
+            Crosshair.SPRITESHEET,
+            DialogBubble.SPRITESHEET,
+            OverheadIcon.SPRITESHEET,
+            PapaChad.SPRITESHEET,
+            Projectile.SPRITESHEET,
+            Slingshot.SPRITESHEET,
+            Sun.SPRITESHEET,
+            Sword.SPRITESHEET,
+            RuneDrop.SPRITESHEET,
+            RuneItem.SPRITESHEET,
```
Comment (tclaar) [inline js/classes/asset_manager.js:198]: I think the `RuneDrop` and `RuneItem` spritesheets are (and will likely always be) the same, so we don't need both of these entries.

File: js/classes/entities/foreground/hud.js [hunk 3]
```diff
@@ -258,11 +258,11 @@ class AmmoLabel {
         this.type = type;
         this.label = new ItemLabel(
 
-            pos, 
-            new Animator(AmmoItem.SPRITESHEET, new Vector(0, type * AmmoItem.SPRITESHEET_ENTRY_HEIGHT), 
+            pos,
+            new Animator(AmmoItem.SPRITESHEET, new Vector(0, type * AmmoItem.SPRITESHEET_ENTRY_HEIGHT),
                 AmmoItem.SIZE, 1, 1),
-                inputName,
-                INVENTORY.getAmmo(type).amount
```
Comment (tclaar) [inline js/classes/entities/foreground/hud.js:265]: Why was the indentation here changed?

File: js/classes/entities/midground/chad.js
Comment (tclaar) [inline js/classes/entities/midground/chad.js:1]: The `Chad.SCALE`, `Chad.SCALED_SIZE`, and `Chad.SPEED` static constants are obsolete and should be removed now that we use instance fields to represent these properties. A number of operations (the `createBoundingBox` method, `Chad.BOUNDING_BOX_OFFSET` static constant, block collisions, and maybe others that I missed) still rely on some of these constants and should be updated to use the new instance fields.

**Review by tclaar — APPROVED** [review]
Thanks for cleaning up this mess, Caleb! I believe you said everything works, and I didn't notice anything game-breaking in the code, so I think this is ready to merge. There are some flaws that were inherited from previous PRs but I don't think there's anything critical. 

Good work!

**Review by b1gd3vd0g — APPROVED** [review]

**Pull request merged**